src/main/java/seedu/address/model/entry/Amount.java

```diff
@@ -40,8 +44,19 @@ public double getValue() {
         return amount;
     }
 
+    /**
+     * Adds the value in the two specified {@code Amount} and returns a new {@code Amount}
+     * that contains a value equivalent to the sum. The addition respects the standard
+     * addition operator and its properties, such as commutativity, associativity, identity
+     * and distributivity.
+     *
+     * @param amount1 An {@code Amount} containing the value to be added
+     * @param amount2 An {@code Amount} containing the value to be added
+     * @return An {@code Amount} containing the value of the sum of the two specified {@Amount}.
+     */
     public static Amount add(Amount amount1, Amount amount2) {
-        return new Amount(String.valueOf(amount1.getValue() + amount2.getValue()));
+        double amount = amount1.getValue() + amount2.getValue();
+        return new Amount(AMOUNT_FORMAT.format(amount));
```

Comment:
Need to format the string before passing into the constructor, in order to ensure the string has exactly 2 decimal digits, which is not guaranteed.

Reply:
Is this formatting causing the test cases to fail?

Reply:
Thanks for the remainder I thought it was something wrong with the codecov. I tried to look into it and I realized the issue was a result of tightening the regex for the `Amount` class. When we store a double value, such as `1.00` the value in the double is `1.0` and does not preserve the precision that we specify. In some cases of the code, we call `Amount.getValue` which returns a double and we expect it to return `1.00`, such as in the code below, causing the regex validation to fail.

![Screenshot 2022-10-27 at 7 28 11 AM](https://user-images.githubusercontent.com/21200681/198157501-d81adabf-c4e5-4c79-87f2-39647c75f657.png)

![Screenshot 2022-10-27 at 7 31 02 AM](https://user-images.githubusercontent.com/21200681/198157846-fe365000-1f91-4769-9575-423b50e759e3.png)

I think there are 2 ways of fixing this:
- Try to use `BigDecimal` class, but this would affect changing quite a bit of files especially because we used the amount value as a `double` quite a lot. Although we can preserve accuracy in calculations.
- Loosen the validation regex to accept 1 or 2 decimal place as before, but then inputs like `5.1` would become valid which is not ideal.

Reply:
Hey sorry for the late reply, is it possible to complete 1? otherwise we can shift it as a bug fix after PE tmr!

Reply:
Okay!